Add a sigstore signing parameter file format, and CLI utility (#1787)
containers/image: docs/containers-sigstore-signing-params.yaml.5.md

% CONTAINERS-SIGSTORE-SIGNING-PARAMS.YAML 5 sigstore signing parameters Man Page
% Miloslav Trmač
% January 2023

# NAME
containers-sigstore-signing-params.yaml - syntax for the sigstore signing parameter file

> # DESCRIPTINO

Reviewer: ```suggestion
# DESCRIPTION
```

> Sigstore signing parameter files are used to store options that may be required to create sigstore signatures.

Reviewer: I thought this was going to be a single file?  If so, suggest:
```suggestion
The sigstore signing parameter file is  used to store options that may be required to create sigstore signatures.
```

Reviewer: Ignore, I see where you were getting at down below.

> There is no standard location for these files; they are user-managed, and used as inputs to a container image signing operation,

Reviewer: On the fence, your call.  "standard location" or "default location"?

Author: (I’m not very sure that sentence even needs to exist… I just wanted to give _some_ context for users who stumble on the page directly, without starting at a `--sign-by-sigstore` option documentation.)

Sure, “default location” works.

   Optional: if this is not set, the user must provide the passphrase interactively.

### Signing with Fulcio-generated Certificates

Instead of a static private key, the signing process generates a short-lived key pair
and requests a Fulcio server to issue a certificate for that key pair,
based on the user authenticating to an OpenID Connect provider.

To specify Fulcio, include a `fulcio` sub-object with one or more of the following keys.
In addition, a Rekor server must be specified as well.

- `fulcioURL:` _URL_

  Required. URL of the Fulcio server to use.

- `oidcMode:` `interactive` | `deviceGrant` | `staticToken`

  Required. Specifies how to obtain the necessary OpenID Connect credential.

  `interactive` opens a web browser on the same machine, or if that is not possible,
  asks the user to open a browser manually and to type in the provided code.
  It requires the user to be able to directly interact with the signing process.

  `deviceGrant` uses a device authorization grant flow (RFC 8628).
  It requires the user to be able to read text printed by the signing process, and to act on it reasonably promptly.

>   `staticToken` provides a pre-existing Open ID Connect ID token which must have been obtained separately.

Reviewer: Based solely on grokking Google, it looks like it should be "OpenID Connect".  (https://openid.net/connect/).  Please change here and elsewhere.  I've added quotes to set it off from the rest of the text and to let the reader know that's a thing that could be looked up if they're unfamiliar as I was.  That said, I'm still on the fence about having the quotes, so no foul if you want to drop them.

```suggestion
  `staticToken` provides a pre-existing "OpenID Connect" ID token which must have been obtained separately.
```

Reviewer: I'll hop off the fence, drop the quotes.  I see you have a number of instances and it would look odd all over.

Author: Thanks! The quotes are a great idea for separating the two terms.

(I didn’t find any other instance of “Open ID” with a space.)

Author: How about quoting the “ID token” part instead? That would be less useful as a stand-alone search term, but assuming the readers figure out `OpenID Connect` is a term to search from other parts of the page, making it clear “ID token” is a single phrase might be worthwhile.

Reviewer: Works for me!

Author: Thanks, changed to quote “ID token”.
